### PR TITLE
Add verified contracts list page

### PR DESCRIPTION
**Pull request by sl1depengwyn**
Resolves https://github.com/blockscout/blockscout/issues/5576
Resolves https://github.com/blockscout/blockscout/issues/6287

## Changelog

- Added verified contract list page on address `/verified-contracts`
- Added contracts counters to hold number of all contracts, number of all contracts appeared in last 24 hours, number of verified contracts, number of verified contracts verified in last 24 hours. Those are responsible for displaying statistics on the verified-contracts page
- Changed nav bar in order to put there verified contracts in reasonable place (check the screenshot)
image

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [x] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

**Review by nikitosing — COMMENTED** [review]
for some reasons some contracts have infinite spinner

image

File: apps/block_scout_web/lib/block_scout_web/chain.ex [hunk 1]
```diff
@@ -216,6 +217,10 @@ defmodule BlockScoutWeb.Chain do
     [paging_options: %{@default_paging_options | key: {name, type, value}}]
   end
 
+  def paging_options(%{"smartcontract_id" => id}) do
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/chain.ex:220]: Nitpick: let's split words by underscore: `smart_contract_id`

**Review by vbaranov — COMMENTED** [review]
@sl1depengwyn since you moved counters to /chain/cache directory, I think it makes more sense to move tests to the same sub-path. What do you think?

**Review by vbaranov — APPROVED** [review]

File: apps/explorer/lib/explorer/chain.ex [hunk 2]
```diff
@@ -6479,4 +6502,99 @@ defmodule Explorer.Chain do
...
+  def verified_contracts(options \\ []) do
+    paging_options = Keyword.get(options, :paging_options, @default_paging_options)
+    necessity_by_association = Keyword.get(options, :necessity_by_association, %{})
+    filter = Keyword.get(options, :filter, nil)
+    search_string = Keyword.get(options, :search, nil)
+
+    query = from(contract in SmartContract, select: contract, order_by: [desc: :inserted_at])
+
+    query
+    |> filter_contracts(filter)
```
Comment (nikitosing) [inline apps/explorer/lib/explorer/chain.ex:6518]: filter does not work with search

image

File: apps/block_scout_web/lib/block_scout_web/controllers/verified_contracts_controller.ex [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  import BlockScoutWeb.Chain,
+    only: [paging_options: 1, next_page_params: 3, split_list_by_page: 1, fetch_page_number: 1]
+
+  alias BlockScoutWeb.{Controller, VerifiedContractsView}
+  alias Explorer.{Chain, Market}
+  alias Explorer.ExchangeRates.Token
+  alias Phoenix.View
+
+  @necessity_by_association %{[address: :token] => :required}
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/controllers/verified_contracts_controller.ex:12]: I think here you don't need :required, :optional will be enough, isnt it?

File: apps/block_scout_web/lib/block_scout_web/templates/verified_contracts/_metatags.html.eex [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<title>
+  <%= gettext(
+        "Verified contracts - %{subnetwork} Explorer",
+        subnetwork: BlockScoutWeb.LayoutView.subnetwork_title()
+    ) %>
+</title>
+<meta name="keywords" content="<%= "Verifed contracts, #{BlockScoutWeb.LayoutView.subnetwork_title()}, #{Explorer.coin()}" %>">
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/templates/verified_contracts/_metatags.html.eex:7]: Let's add gettext for this line also

**Review by nikitosing — CHANGES_REQUESTED** [review]

**Review by nikitosing — APPROVED** [review]

**Pull request merged**